lib/tasks/multitenant/tenants.rake

```diff
@@ -77,6 +77,49 @@ namespace :multitenant do
       Rails.logger.error "Inconsistent tenant_ids for:\n#{inconsistent.map {_1.join(" ")}.join("\n")}"
     end
 
+    desc 'Check and remove orphaned objects (whose tenant is missing), pass "destroy" argument to delete'
+    task :cleanup_orphans, [:mode] => :environment do |_task, args|
+      destroy = args[:mode] == "destroy"
+
+      puts "Checking orphaned objects..."
+      puts "WARNING: the found orphan objects will be destroyed" if destroy
+
+      provider_account_ids = Account.where(provider: true).pluck(:id) + [Account.master.id]
+
+      base_models.each do |model|
+        orphaned_objects = model.where.not(tenant_id: provider_account_ids)
+
+        if orphaned_objects.exists?
+          puts "Found #{orphaned_objects.size} orphaned objects for model #{model.name}:"
+          orphaned_objects.find_in_batches(batch_size: 100).with_index do |batch, index|
+            puts "Processing batch #{index+1} of model #{model.name}..."
+            batch.each do |object|
+              puts "- ID: #{object.id}, Tenant ID: #{object.tenant_id}"
+              DeletePlainObjectWorker.perform_later(object) if destroy
+            end
+          end
+          orphaned_objects.find_each { |obj|  }
+
+
```

Comment:
A nitpick: why two empty lines here?

Reply:
LOL, I'm more interested in why there is an empty block there :sweat_smile: 

I think I might have pushed something half-finished... or half-started

Reply:
Oh I missed that one 🤕